pywinauto/base_application.py

```diff
@@ -242,22 +291,189 @@ def __resolve_control(self, criteria, timeout=None, retry_interval=None):
+    def wait_not(self, wait_for, timeout=None, retry_interval=None):
+        """
+        Wait for the window to not be in a particular state/states.
+        :param wait_for_not: The state to wait for the window to not be in. It can be any
+            of the following states, also you may combine the states by space key.
+             * 'exists' means that the window is a valid handle
+             * 'visible' means that the window is not hidden
+             * 'enabled' means that the window is not disabled
+             * 'ready' means that the window is visible and enabled
+             * 'active' means that the window is active
+        :param timeout: Raise an :func:`pywinauto.timings.TimeoutError` if the window is sill in the
+            state after this number of seconds.
+            Default: :py:attr:`pywinauto.timings.Timings.window_find_timeout`.
+        :param retry_interval: How long to sleep between each retry.
+            Default: :py:attr:`pywinauto.timings.Timings.window_find_retry`.
+        An example to wait until the dialog is not ready, enabled or visible: ::
+            self.Dlg.wait_not("enabled visible ready")
+        .. seealso::
+            :func:`WindowSpecification.wait()`
+            :func:`pywinauto.timings.TimeoutError`
+        """
+
+        warnings.warn("Wait_not method is deprecated and will be removed. "
+                      "You can use not_exists() instead of wait_not('exists'). "
+                      "wait_not_visible(), wait_not_enabled() and wait_not_active() are methods of "
+                      "HwndWrapper object, so you can use it like .find().wait_not_active(), "
+                      ".find().wait_not_visible().wait_not_enabled(), etc.")
+
+        if timeout is None:
+            timeout = Timings.window_find_timeout
+        if retry_interval is None:
+            retry_interval = Timings.window_find_retry
+        correct_wait_for = wait_for.lower().split()
+        if 'ready' in correct_wait_for:
+            correct_wait_for.remove('ready')
+            if 'visible' not in correct_wait_for:
+                correct_wait_for.append('visible')
+            if 'enabled' not in correct_wait_for:
+                correct_wait_for.append('enabled')
+
+        if 'exists' in correct_wait_for:
+            if not self.not_exists(timeout, retry_interval):
+                raise TimeoutError("Object with the given criteria {} still exists".format(self.criteria[-1]))
+
+        else:
+            time_left = timeout
+            start = timestamp()
+            try:
+                ctrl = self.find(time_left, retry_interval)
+            except (findwindows.ElementNotFoundError,
+                    findbestmatch.MatchError,
+                    controls.InvalidWindowHandle,
+                    controls.InvalidElement,
+                    TimeoutError) as e:
+                warnings.warn("Object with the given criteria {} does not exist".format(self.criteria[-1]))
```

Comment:
Why do we need this warning? I guess it's normal return without any warning. Is it left as debug step? Please remove it.